lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -423,8 +423,17 @@ void LayoutPropagation::rewriteRegion(Region &region) {
           auto it = layouts.find(operand.get());
           if (it == layouts.end())
             continue;
-          Attribute encoding =
-              cast<RankedTensorType>(operand.get().getType()).getEncoding();
+
+          Attribute encoding;
+          if (isa<triton::nvidia_gpu::TMEMStoreOp>(&op)) {
```

Comment:
Only certain layouts are compatible with storing to tmem, so in the general case I wouldn't expect this to work. I'm surprised tmem store isn't a layout anchor though. Perhaps propagating the layout backward from the tmem store would fix your issues?